src/js/core.js

```diff
@@ -490,6 +490,10 @@ function MediumEditor(elements, options) {
             return this.options.ownerDocument.execCommand('insertImage', false, this.options.contentWindow.getSelection());
         }
 
+        if (action.match(/h(1|2|3)/)) {
```

Comment:
If you change your keyboard commands to call `append-h1`, `append-h2`, etc. Then you don't need this new line of code.

In `execAction`, you can see that the code will automatically take any command that starts with `append-` and call `formatBlock` with it.